### PR TITLE
ISAICP-2741: Compile SASS to CSS on build

### PR DESCRIPTION
**Pull request by pfrenssen**
```
                                                               .---.
                                                              /  .  \
                                                             |\_/|   |
                                                             |   |  /|
  .----------------------------------------------------------------' |
 /  .-.                                                              |
|  /   \   * I accept the conditions in the Joinup Individual        |
| |\_.  |    Contributor Assignment Agreement.                       |
|\|  | /|                                                            |
| `---' |  * I accept the terms laid out in the Code of Conduct.     |
|       |                                                           /
|       |----------------------------------------------------------'
\       |
 \     /
  `---'
```

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -416,48 +412,60 @@
...
-            <not>
-                <available file="/home/cphp" type="dir" property="environment.cphp" />
-            </not>
-            <then>
-                <fail message="Only run this target on a ContinuousPHP test environment" />
-            </then>
-        </if>
-        <echo message="Put the odbc configuration file in the home folder where CPHP can find it"/>
+    <target name="setup-continuousphp"
+            depends = "check-continuousphp">
```
Comment (brummbar) [inline build.xml]: Shouldn't this be `depends="...` without spaces?

File: build.xml [hunk 2]
```diff
@@ -416,48 +412,60 @@
...
+                <available file="/home/cphp" type="dir" property="environment.cphp" />
+            </not>
+            <then>
+                <fail message="Only run this target on a ContinuousPHP test environment" />
+            </then>
+        </if>
+    </target>
+
+    <target name="build-continuousphp"
+            depends = "check-continuousphp">
```
Comment (brummbar) [inline build.xml]: Same here, maybe without spaces?

File: web/themes/joinup/.gitignore [hunk 3]
```diff
@@ -1,4 +1,6 @@
 # Ignore generated directories
+css
+prototype/css/*.css
```
Comment (brummbar) [inline web/themes/joinup/.gitignore]: It should be enough to have these lines in the main .gitignore file.

File: web/themes/joinup/prototype/sass/_mixins.sass [hunk 4]
```diff
@@ -31,7 +31,6 @@
     @media (min-width: $min-width) and (max-width: $max-width)
       @content
 
-  // Else render only min breakpoint
```
Comment (brummbar) [inline web/themes/joinup/prototype/sass/_mixins.sass]: Maybe we can keep the comment by moving it inside the else ?

File: build.xml [hunk 5]
```diff
@@ -416,48 +412,60 @@
...
+        <echo message="Install the SASS compiler" />
+        <exec dir="${project.basedir}"
+              command="gem install sass"
```
Comment (brummbar) [inline build.xml]: We could also specify `-v 3.4.22` to pin it to a version.

Comment (pfrenssen) [inline build.xml]: If you want to pin the version it should be done in a Gemfile not by hard coding it in a build target.

Comment (brummbar) [inline build.xml]: We discussed this in person and agreed that pinning is not necessary, as the chances of differences introduced by a new version are really limited.

**Comment by claudiu-cristea** [comment]
Reviewed last commits: 9aee0da, 0c90832, cc0eda, a21917e.

**Pull request merged**